gufe/transformation.py

```diff
@@ -0,0 +1,51 @@
+from .protocols import Protocol
+
+
+class Transformation:
+    """An edge of an alchemical network.
+
+    Connects two chemical states, with directionality.
+
+    Attributes
+    ----------
+    protocol : Protocol
+        The protocol used to perform the transformation.
+        Includes all details needed to perform calculation and encodes the
+        alchemical pathway used.
+
+    """
+
+    def __init__(
+            self,
+            protocol
+            ):
+
+        self._protocol = protocol
+
+
+    # these are ambiguous; will need a better way to consider this
+    # these could be expensive operations
+    def dg(self, estimator=None):
+        """Free energy difference of transformation based on given estimator,
+        using only data available for this edge.
+
+        """
+        ...
+
+    def ddg(self, estimator=None):
+        ...
```

Comment:
We're thinking of splitting pre/post computation data structures.  Maybe `dg` and `ddg` are best as methods of the edges of results network?

Reply:
That or I would say that we should just conflate them into a `results` method. My reasoning here is that if `Transformation` is meant to be the parent class to several downstream classes, then it should be up to children classes to implement various estimators - not everything might fit to every method?